.github/workflows/ci.yml

```diff
@@ -38,3 +34,12 @@ jobs:
         run: cargo test --all
       - name: Run tests with toggle feature
         run: cargo test --all --features toggle
+
+  alls-green:
+    if: always() && (github.event_name != 'pull_request' || github.event.pull_request.head.repo.full_name != github.event.pull_request.base.repo.full_name)
+    runs-on: ubuntu-latest
+    needs:
+      - lint
+      - test
+    steps:
+      - run: ${{ !contains(needs.*.result, 'failure') }}
```

Comment:
It would be fine to use alls-green(the action) here, since we use github's actions runners, so the action can just default to using a python installation from the host, without pulling a docker image, but I don't think we need any of it's more advanced features, so I prefer this